src/store/models/models.reducers.ts

```diff
@@ -103,21 +103,34 @@ const rootSelector = (state: RootState): ModelsState => state.models;
 const selectModelsIsLoading = createSelector([rootSelector], (state) => {
   return state.isLoading;
 });
+
 const selectModelsError = createSelector([rootSelector], (state) => {
   return state.error;
 });
+
 const selectModels = createSelector([rootSelector], (state) => {
   return state.models;
 });
+
+const selectModel = createSelector(
+  [selectModels, (_state, modelId: string) => modelId],
+  (models, modelId) => {
+    return models.find(({ id }) => id === modelId) as OpenAIEntityModel;
```

Comment:
you can use modelsMap instead
Moreover it's possible that model will not be found, but because of your type assertion it will be not catched

Reply:
Implemented selector with modelsMap.
Additional check added